app/controllers/game_board_controller.rb

```diff
@@ -128,25 +133,30 @@ def make_suggestion
       room = params[:location_id]
 
       results = []
-
-      location_id = $game.game_board.rooms.collect(:name == room).first.id
-
-      $game.get_players.each do |p|
-        #If this player is the one that is in the suggestion, move him to the room
-        if p.get_board_piece.name == suspect
-          @game.game_board.move_player(p, location_id)
+      
+      location_id = nil
+      
+      $game.game_board.rooms.collect{|k,v| v}.each do |r|
+        if r.name == room
+          location_id = r.id
+          break
         end
-
-        if p.get_cards[:weapon].name == weapon
-          results.push({:player => p, :card => weapon})
+      end
+      
+      $game.get_players.collect { |k, v| v }.each do |p|
```

Comment:
Ok, I just grabbed the method that was used in the other methods. That way
looks a lot cleaner though.

On Thu, Dec 10, 2015 at 2:41 PM, Billy Yarosh <notifications@github.com>
wrote:

> In app/controllers/game_board_controller.rb
> <https://github.com/en-604-pionNeers/clueLess-server/pull/6#discussion_r47274164>
> :
>
> >          end
> > -
> > -        if p.get_cards[:weapon].name == weapon
> > -          results.push({:player => p, :card => weapon})
> > +      end
> > +
> > +      $game.get_players.collect { |k, v| v }.each do |p|
>
> I think you can do:
>
> $game.get_players.each do |index, player|
> end
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/en-604-pionNeers/clueLess-server/pull/6/files#r47274164>
> .
>